Ingress/controllers/nginx-third-party/nginx/utils.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nginx
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+func skydnsIPPort(kubeClient *unversioned.Client) (ipPort string) {
+	glog.Info("Waiting for skydns service")
+	wait.Poll(1*time.Second, 5*time.Minute, func() (bool, error) {
+		svc, err := kubeClient.Services("kube-system").Get("kube-dns")
+		if err != nil {
+			glog.Errorf("error queriyng service kube-system/kube-dns: %v", err)
+			return false, nil
+		}
+
+		ipPort = fmt.Sprintf("%v:%v", svc.Spec.ClusterIP, 53)
+		return true, nil
+	})
+
+	return
+}
+
+func getService(kubeClient *unversioned.Client, name string) k8sService {
+	// Wait for the default backend Service. There's no pretty way to do this.
```

Comment:
does this function only handle the default backend? if so please rename, if not please make it generic